File: solutions/java/2799-number-of-employees-who-met-the-target.java
class Solution {
    public int countCompleteSubarrays(int[] nums) {
        int n = nums.length;
        Set<Integer> set = new HashSet<>();
        for (int num : nums) set.add(num);
        int distinct_elements = set.size();
        int count = 0;
        int left = 0;
        int right = 0;
        Map<Integer, Integer> counter = new HashMap<>();

        while (right < n) {
            counter.put(nums[right], counter.getOrDefault(nums[right], 0) + 1);
            while (counter.size() == distinct_elements) {
                counter.put(nums[left], counter.get(nums[left]) - 1);
                if (counter.get(nums[left]) == 0) {
                    counter.remove(nums[left]);
                }
                left++;
                count += n - right;
            }
            right++;
        }
        return count;        
    }
}

File: solutions/javascript/2799-number-of-employees-who-met-the-target.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countCompleteSubarrays = function(nums) {
    let n = nums.length;
    let distinct_elements = new Set(nums).size;
    let count = 0;
    let left = 0;
    let right = 0;
    let counter = new Map();

    while (right < n) {
        counter.set(nums[right], (counter.get(nums[right]) || 0) + 1);
        while (counter.size === distinct_elements) {
            counter.set(nums[left], counter.get(nums[left]) - 1);
            if (counter.get(nums[left]) === 0) {
                counter.delete(nums[left]);
            }
            left++;
            count += n - right;
        }
        right++;
    }
    return count;    
};
